### PR TITLE
feature: add support for check_mode to grafana_dashboard

### PR DESCRIPTION
**Pull request by optica-phoffmann**
##### SUMMARY
I added support for check_mode to grafana_dashboard.

I think I covered all uses. With check_mode a dashboard
 - is not deleted
 - is not updated
 - is not created
 - is not exported (I think this might be debatable)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
grafana_dashboard

### CODE REVIEW

**Comment by rndmh3ro** [comment]
Thanks, nice addition!

Some things:

- can you please add a changelog fragment?
- can you add the following code to the file ` tests/integration/targets/grafana_dashboard/runme.sh`:
```
diff --git a/tests/integration/targets/grafana_dashboard/runme.sh b/tests/integration/targets/grafana_dashboard/runme.sh
index 867afb0..9daa487 100755
--- a/tests/integration/targets/grafana_dashboard/runme.sh
+++ b/tests/integration/targets/grafana_dashboard/runme.sh
@@ -2,4 +2,5 @@

 set -eux

+ansible-playbook site.yml --check
 ansible-playbook site.yml
  ```

File: plugins/modules/grafana_dashboard.py [hunk 1]
```diff
@@ -415,7 +415,13 @@ def grafana_create_dashboard(module, data):
...
 
     if dashboard_exists is True:
-        if grafana_dashboard_changed(payload, dashboard):
+        grafana_dashboard_changed = is_grafana_dashboard_changed(payload, dashboard)
+
+        if grafana_dashboard_changed:
+            if module.check_mode:
+                module.exit_json(failed=False,
+                                 changed=True,
+                                 msg="Dashboard %s will be updated" % payload['dashboard']['title'])
```
[the remark this answers is not in the document]
Comment (optica-phoffmann) [reply plugins/modules/grafana_dashboard.py:424]: I could not find an _official_ guidance about check_mode messages. I understand testing is easier.

As an end user I'd like to be able to tell run mode and check mode apart by the actual module message. I don't know if it still the case but I remember times when running Ansible jobs in AWX had no clear indication in the Web UI whether the job was executed in run mode or check mode.

[the remark this answers is not in the document]
Comment (rndmh3ro) [reply plugins/modules/grafana_dashboard.py:424]: Ultimately I am fine with both variants.

From my experience using Ansible, the message is the same, regardless of check-mode or not.
One example:

```
> ansible localhost -m  docker_container -a "name=test image=busybox" -C
[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | CHANGED => {
    "actions": [
        {
            "pulled_image": "busybox:latest"
        },
        {
            "create_parameters": {
                "AttachStderr": true,
                "AttachStdin": false,
                "AttachStdout": true,
                "ExposedPorts": {},
                "Image": "busybox",
                "OpenStdin": null,
                "StdinOnce": false
            },
            "created": "Created container"
        }
    ],
    "changed": true
}
```

If you want to keep the different messages, that's okay! However you'd then have to change the tests. They assume always the same result message:

```
TASK [../../grafana_dashboard : ansible.builtin.assert] ************************
fatal: [localhost]: FAILED! => {
    "assertion": "result.msg == 'Dashboard Zabbix Host Status created'",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
``

**Comment by optica-phoffmann** [comment]
For writing tests, there was a lot more to do than simple adding a `ansible-playbook site.yml --check` as most of the current tests were based on one another.

**Review by rndmh3ro — APPROVED** [review]

**Comment by rndmh3ro** [comment]
Thank you for this great addition!

**Pull request merged**